apps/ecommerce/frontend/src/hooks/useExternalResource.tsx

```diff
@@ -30,13 +31,15 @@ const useExternalResource = (resource?: ExternalResourceLink) => {
         {
           'x-contentful-data-provider': resourceProvider.toLowerCase(),
           'X-Contentful-Data-Provider-BaseURL': sdk.parameters.instance.baseUrl,
+          'x-contentful-shopify-domain': apiEndpoint,
```

Comment:
Probably a question for the future and not for this PR, but I am curious how we are going to handle different providers using this approach. I guess maybe use information from the config to construct the headers rather than hardcoding them?

Reply:
True, we are aiming for the universal frontend being agnostic of the provider, I can try generalizing this more.

Reply:
Poked around at this and seems like it might be more work out of scope for this ticket. Would it be okay to address as a follow-up?

Reply:
Yes, that makes sense!